check.ml

```diff
@@ -0,0 +1,118 @@
+open Instr
+
+exception MissingMain
+exception DuplicateFunctionDeclaration of identifier
+exception InvalidFunctionDeclaration of identifier
+exception DuplicateVersion of identifier * label
+exception EmptyFunction of identifier
+exception DuplicateParameter of identifier * variable
+
+exception ErrorAt of identifier * label * exn
```

Comment:
Minor detail, but I didn't find it obvious why this was called `ErrorAt` and why only certain exceptions are wrapped by `ErrorAt`. Only by looking at usages of `ErrorAt` did I realize it's for errors at a specific function and version (as opposed to general errors in the program).

I think a short comment here would help.